inc/add_site_ui/namespace.php

```diff
@@ -186,9 +188,46 @@ function output_add_site_page() {
  * @return bool True if valid, false otherwise.
  */
 function validate_domain_segment( string $segment ) : bool {
+	$segment = sanitize_domain_segment( $segment );
```

Comment:
I'm not keen on this being here if the function is for validating a value - it shouldn't really be modifying the value as well. I would expect the value from user input to be sanitised first and then the value of that validated.

I'm guessing that validation was failing in some cases without it so it might be that you either need to sanitize the user input early before validating or perhaps you might need to split `$segment` on `.` and validate each part of the domain.